### ISSUE TITLE
[Ray Collective Lib] Enable CI 

### ISSUE THREAD

**Issue by jovany-wang** [ray-project/ray#25396]
### What happened + What you expected to happen

All cases in ray collective lib are disabled, we should enable them since we have been developing and maintaining it.

### Versions / Dependencies

1

### Reproduction script

1

**Comment by simon-mo** [comment]
Good point. @zhisbug we now have GPU CI, do you think it make sense to include it in our CI and other test cases? Any pointers?

**Comment by jovany-wang** [comment]
reassigning...

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
